### ISSUE TITLE
External import statements in swiftinterface doesn't respect -fmodule-map-file compiler option

### ISSUE THREAD

**Issue by Kudo** [apple/swift#61335]
**Describe the bug**

I came across a compiler issue for `.swiftinterface` with external import dependencies. The dependency is an Objective-C module with custom modulemap. Typically, we can use the `-fmodule-map-file` compiler option to specify custom modulemap. This is also heavily used in CocoaPods. However, `.swiftinterface` module resolution seems not support this option.

**Steps To Reproduce**

To demonstrate the issue, I have a minimal reproducible example at https://github.com/Kudo/Swift-Module-Issue . Although it's getting little complicated, if there's anything unclear, please let me know. Thanks!

**Expected behavior**

[the case `App -> ModuleContainer (swiftinterface) -> ModuleB` in the repro](https://github.com/Kudo/Swift-Module-Issue#result-of-module-resolutions) should not have build errors.

**Screenshots**

screenshot from [the repro](https://github.com/Kudo/Swift-Module-Issue#result-of-module-resolutions)

![image](https://user-images.githubusercontent.com/46429/192824084-eb101c7a-7c98-4e8f-b941-62791512e52f.png)

**Environment (please fill out the following information)**

 - OS: macOS 12.6
 - Xcode Version/Tag/Branch: Xcode 14.0.1 (14A400) 
 - Apple Swift version 5.7 (swiftlang-5.7.0.127.4 clang-1400.0.29.50)

**Additional context**

same as https://github.com/apple/swift/issues/58268 but tested on xcode 14. it is still broken.

**Comment by maciejburda** [comment]
@Kudo Did you fix it by any chance?

**Comment by Kudo** [comment]
@maciejburda no, i don't know much for swift compiler internally and have no context for the fix.
